### PR TITLE
add TLS/SSL options for connecting to MQTT server

### PR DESCRIPTION
**Pull request by andreasbrett**
Quick implementation of what @Diaoul requested in https://github.com/Koenkk/zigbee2mqtt/issues/1700.

### CODE REVIEW

File: lib/mqtt.js [hunk 1]
```diff
@@ -20,6 +20,16 @@ class MQTT {
...
         };
+        
+        if (mqttSettings.key && mqttSettings.cert && mqttSettings.ca) {
+            logger.debug(`Using SSL/TLS settings for connection:`);
+            logger.debug(` - key path = ${mqttSettings.key}`);
+            logger.debug(` - cert path = ${mqttSettings.cert}`);
+            logger.debug(` - CA cert path = ${mqttSettings.ca}`);
+            options.key = mqttSettings.key;
+            options.cert = mqttSettings.cert;
+            options.ca = mqttSettings.ca;
```
Comment (Diaoul) [inline lib/mqtt.js]: Don't you need to read the file like in [the example](https://github.com/mqttjs/MQTT.js/blob/master/examples/tls%20client/mqttclient.js#L20) such that `options.key` and friends contain the *contents* of the files and not the path? Or is that done in a previous step?

Comment (andreasbrett) [reply lib/mqtt.js]: You're absolutely right. Thought mqtt.js would load them but thanks for your hint. Fixed now.

**Comment by andreasbrett** [comment]
@Koenkk Since @Diaoul will only be able to test in a couple of weeks I tested it myself. Good opportunity to up the security of my MQTT server. Hardest thing was to re-configure MQTT but once that was done connecting via z2m worked right away.

PR can be merged!

**Comment by Koenkk** [comment]
Thanks!

**Pull request merged**